pkg/service/system/route53.go

```diff
@@ -129,3 +182,194 @@ func AddRoute53Record(ctx context.Context, dnsName, recordName, regionName strin
+	}
+
+	return nil
+
+}
+
+// DeleteRoute53Records deletes the records from the zone. If a record does not have an ID,
+// it will be looked up. It returns the records that were deleted.
+func DeleteRoute53Records(ctx context.Context, records []types.Route53Record) ([]types.Route53Record, error) {
+
+	hostedZoneId := config.Get().AwsRoute53HostedZoneID
+
+	// Creating AWS Route53 session
+	route53Client, err := getRoute53Sess(defaultRegion)
+	if err != nil {
+		return nil, errors.Wrap(err, "DeleteRoute53Records: failed to create route53 session")
+	}
+
+	hostedZone, err := getHostedZoneFromZoneId(route53Client, hostedZoneId)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "DeleteRoute53Records: failed to get route53 hostedZone ")
+	}
+
+	var deletedRecords []types.Route53Record
+
+	for _, record := range records {
+		err := deleteRoute53Record(ctx, route53Client, record, hostedZoneId, hostedZone)
+		if err != nil {
+			return deletedRecords, err
+		}
+		deletedRecords = append(deletedRecords, record)
+	}
+
+	return deletedRecords, nil
+}
+
+func createRoute53Record(ctx context.Context, route53Client *route53.Route53, zoneID string, record types.Route53Record, hostedZone string) (types.Route53Record, error) {
+	// AWS Route53 TXT record value must be enclosed in quotation marks on create
+	if record.Type == route53.RRTypeTxt {
+		record.Value = strconv.Quote(record.Value)
+	}
+
+	createInput := &route53.ChangeResourceRecordSetsInput{
+		ChangeBatch: &route53.ChangeBatch{
+			Changes: []*route53.Change{
+				{
+					Action: aws.String("CREATE"),
+					ResourceRecordSet: &route53.ResourceRecordSet{
+						Name: aws.String(absoluteName(record.Name, hostedZone)),
+						ResourceRecords: []*route53.ResourceRecord{
+							{
+								Value: aws.String(record.Value),
+							},
+						},
+						TTL:  aws.Int64(int64(record.TTLInSeconds)),
+						Type: aws.String(record.Type),
+					},
+				},
+			},
+		},
+		HostedZoneId: aws.String(zoneID),
+	}
+
+	err := applyChange(ctx, route53Client, createInput)
+	if err != nil {
+		return record, err
+	}
+
+	return record, nil
+}
+
+// AppendRoute53Records adds records to the zone. It returns the records that were added.
+func AppendRoute53Records(ctx context.Context, records []types.Route53Record) ([]types.Route53Record, error) {
+
+	hostedZoneId := config.Get().AwsRoute53HostedZoneID
+
+	// Creating AWS Route53 session
+	route53Client, err := getRoute53Sess(defaultRegion)
+	if err != nil {
+		return nil, errors.Wrap(err, "AddRoute53Record: failed to create route53 session")
+	}
+
+	hostedZone, err := getHostedZoneFromZoneId(route53Client, hostedZoneId)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "AddRoute53Record: failed to get route53 hostedZone ")
+	}
+
+	var createdRecords []types.Route53Record
+
+	for _, record := range records {
+		newRecord, err := createRoute53Record(ctx, route53Client, hostedZoneId, record, hostedZone)
+		if err != nil {
+			return createdRecords, err
```

Comment:
expecting to return partial succeeded list ? can we return `nil, err`. If we are expecting some records to be created before errored out, its possible that next one also might succeed.
Should we try all and accumulate error and return all at once ?

Reply:
Fixed with all or none